examples/CMakeLists.txt

```diff
@@ -7,16 +7,39 @@ project(spbench CXX)
 add_executable(spbench spbench.cpp)
 target_link_libraries(spbench Etaler)
 
+option(ETALER_ENABLE_MATPLOTLIB_CPP "Enable Matplotlib CPP integration" ON)
+
 project(tmexample CXX)
 add_executable(tmexample tmexample.cpp)
+
+if(ETALER_ENABLE_MATPLOTLIB_CPP)
+	find_package(PythonLibs 2.7)
+	target_include_directories(tmexample PRIVATE ${PYTHON_INCLUDE_DIRS})
+	target_link_libraries(tmexample ${PYTHON_LIBRARIES})
+endif()
 target_link_libraries(tmexample Etaler)
 
+
+project(tm_visualize CXX)
+link_directories(${CMAKE_CURRENT_SOURCE_DIR}/visualize/LayersVisualizer/build/)
+link_directories(${CMAKE_CURRENT_SOURCE_DIR}/visualize//LayersVisualizer/Easy3D/build/lib)
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/visualize/LayersVisualizer/)
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/visualize/LayersVisualizer/Easy3D)
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/visualize/LayersVisualizer/imgui)
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/visualize/LayersVisualizer ${CMAKE_CURRENT_SOURCE_DIR}/visualize/LayersVisualizer/build/)
+find_package(OpenGL REQUIRED)
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -pthread -g")
```

Comment:
Please use `set_target_properties(${TARGET} PROPERTIES COMPILE_FLAGS ${ADDITIONAL_BUILD_FLAGS})` to get flags for a single target. Though I believe these flags are set in /CMakeLists.txt already. (You don't need -g and -pthread is provided by tbb)

Reply:
NP